### ISSUE TITLE
Only about 200 imported .session files show up in the Session List

### ISSUE THREAD

**Issue by HubJ** [sienori/Tab-Session-Manager#879]
## Steps to reproduce

1. Start browser (FF 92) with clean profile
2. Install Tab Session Manager
3. Follow the steps to import .session files from a prior installation in FF52
4. Point TSM to a folder with about 532 .session files, and click Import
5. Only about 200 show up in the Session List, whether sorted by oldest or newest first

## Expected result

All imported results would appear in the Session List.

## Actual result

Only about 200 show up.

## Platform information
* Platform (OS): Windows 7
* Version of browser: FF 92
* Version of Tab Session Manager: 6.10

## Additional context

If this is a known limit, please limit the import process to the maximum number of files that can be handled.

**Comment by HubJ** [comment]
I just checked, and confirmed that all .session files had actually been copied from the source into the Sessions folder. Furthermore, sessionmanager.sqlite now includes the names of all 500+ sessions. However, using the search feature to look for words in the names of "imported" sessions outside the scope of the 200 or so that display does not produce any results.

**Comment by sienori** [comment]
Check the number next to `All`. Can you see all the imported sessions?
![image](https://user-images.githubusercontent.com/29514424/135762288-7675b889-5d61-454f-8360-356099a14be2.png)

**Comment by HubJ** [comment]
No. The number shown is 206. The proper number would be over 560.

 

  _____

**Comment by sienori** [comment]
Could you please send me the all .session files to my [email](sienori.firefox@gmail.com) so I can identify the problem?
Of course, I will take care of your privacy.

**Comment by HubJ** [comment]
The Sessions folder containing all 560+ files that were imported is about 1.7 GB in size, so you’d need to advise on some other way to get the info to you.

 

As a workaround, it seems unlikely to me that the problem relates to the imported session files themselves. To confirm that, you could use Windows to create a single.session file of minimal size (saving perhaps one tab), and place it in its own folder outside the Firefox profile. Then, selecting it, and using “copy” from the File Explorer context menu to “copy it here”, you’d then have two identical .session files that would be perfectly formed, and have minimal size. If you then repeated that process for all files in that folder, you’d quickly go from having 2 files to 4, 8,16, etc. Soon you’d have a collection of over 512 .session files. I bet the problem will manifest itself on your machine if you then try to import all of them through TSM.

 

Jeff

 

 

  _____
